routes/api/index.js

```diff
@@ -0,0 +1,19 @@
+const express = require('express')
+const usersRoute = require('./users')
+
+const router = express.Router()
+
+router.use('/users', usersRoute)
+
+router.use((req, res, next) => {
```

Comment:
Can you tell me how this error handling works? I'm not familiar with it and can't find anything online.

Reply:
It's express middleware, express will determine whether the request match pattern string to call middleware. 
So if request have pattern /api/users/..., it will be processed by userRouter. If not, it's will go to 404 handler. We'll add some routers above the 404 handler, and server only run 404 handle middleware when the request not match any routes.
We call next(err) on 404 handler to go error handler. Error handle middleware is the special middleware, it have 4 parameter. When we call next(err), it will go error handler. So all error of API will be processed in this error handler.